_posts/blog/2017-07-18-dwh-part2-we-need-a-storage.md

```diff
@@ -0,0 +1,181 @@
+S3 serves as our data lake. This enables us to only load relevant data selectively to our main storage without losing information, 
+because we could re-load each information afterwards. This reduces costs (S3 is cheap, our main storage is not) and 
+[Amazon Athena](https://aws.amazon.com/de/athena/) even enables to query data in S3 directly.
+
+## The Candidate - AWS Redshift
+
+Most of our infrastructure is hosted on Amazon Web Services (AWS). It would be handy to stay on this well known ground. 
+There must be something out there. And it is indeed. [Redshift](<https://aws.amazon.com/redshift>) is Amazon’s “data warehouse as a service” solution. 
+
+_Amazon Redshift is a fast, fully managed data warehouse that makes it simple and cost-effective to analyze all your data using standard SQL and your existing Business Intelligence (BI) tools._
+
+Basically it is a column oriented database with good support for analytical functions able to handle petabytes of structured data by massively parallel query execution. Sounds good. 
+Fulfills our requirements. Why not giving it a chance?
+
+__Pros:__
+
+* (unlimited) scalability
+* good support for analytical functions
+* accessible like any other database
+* good integration in AWS infrastructure (S3, AWS Lambdas, etc.)
+* good support to import data from S3
+
+__Cons:__
+
+* Costly (depending on the type and number of nodes used)
+* steep learning curve in order to boost performance
+* maybe training necessary (or even consultancy)
+* no native support for semi-structured data
+* database administration is complex (column compression, distribution, vacuumization, …)
+
+## The Opponent - Snowflake
+
+Redshift is not the only data warehouse as a service candidate. [Snowflake](https://www.snowflake.net/) is another one. 
+To be honest, we evaluated Snowflake a while ago. 
+A lot has changed since then, but let’s stick to the data we collected sometime in 2016.
+
+_From zero to petabytes, our elastic data warehouse architecture scales to any volume of data painlessly and inexpensively. 
+No more painful decisions about what data you can afford to keep._
+
+__Pros:__
+
+* storage is cheap (maybe) -> see cons
+* native support for semi-structured data
+* (unlimited) scalability
+* good integration into AWS infrastructure (S3, AWS Lambdas, etc.)
+* database administration is done automatically
+
+__Cons:__
+
+* nontransparent pricing policy
+* missing information about tooling (ETL)
+
+Snowflake created a database layer on top of S3 (a little bit like Amazon Athena did later). Unfortunately it wasn’t easy to really test Snowflake at that time. 
+Consequently we did not investigate further. Maybe we would handle this issue differently today.
+
+## Finally
+
+We decided for Redshift in the end. I have to admit it wasn’t a too unbiased process, because more or less we wanted to try Redshift anyway. 
```

Comment:
I have to admit it wasn’t a too unbiased process -> I have to admit the process was a bit biased, because we wanted to try it anyway.